src/drivers/safety_button/safety.cpp

```diff
@@ -0,0 +1,330 @@
+
+		} else {
+			pattern = LED_PATTERN_IO_ARMED;
+		}
+
+	} else if (_armed.armed) {
+		pattern = LED_PATTERN_FMU_ARMED;
+
+	} else {
+		pattern = LED_PATTERN_FMU_OK_TO_ARM;
+
+	}
+
+	/* Turn the LED on if we have a 1 at the current bit position */
+	px4_arch_gpiowrite(GPIO_LED_SAFETY, !(pattern & (1 << blink_counter++)));
+
+	if (blink_counter > 15) {
+		blink_counter = 0;
+	}
+}
+
+int
+SafetyButton::task_spawn(int argc, char *argv[])
+{
+	if (PX4_MFT_HW_SUPPORTED(PX4_MFT_PX4IO)) {
+		PX4_ERR("not starting (use px4io for safety button)");
+
+		return PX4_ERROR;
+
+	} else if (circuit_breaker_enabled("CBRK_IO_SAFETY", CBRK_IO_SAFETY_KEY)) {
+		PX4_WARN("disabled by CBRK_IO_SAFETY, exiting");
+		return PX4_ERROR;
+
+	} else {
+		/* schedule a cycle to start things */
+		int ret = work_queue(HPWORK, &_work, (worker_t)&SafetyButton::cycle_trampoline, nullptr, 0);
+
+		if (ret < 0) {
+			return ret;
+		}
+
+		_task_id = task_id_is_work_queue;
+	}
+
+	return PX4_OK;
+}
+
+void
+SafetyButton::cycle_trampoline(void *arg)
+{
+	SafetyButton *dev = reinterpret_cast<SafetyButton *>(arg);
+
+	// check if the trampoline is called for the first time
+	if (!dev) {
+		dev = new SafetyButton();
+
+		if (!dev) {
+			PX4_ERR("alloc failed");
+			return;
+		}
+
+		_object = dev;
+	}
+
+	dev->cycle();
+}
+
+void
+SafetyButton::cycle()
+{
+	const hrt_abstime cycle_timestamp = hrt_absolute_time();
+
+	if (cycle_timestamp - _last_safety_check >= 100_ms) {
+		_last_safety_check = cycle_timestamp;
+
+		/**
+		 * Get and handle the safety status at 10Hz
+		 */
+		safety_s safety = {};
+
+		/* read safety switch input and control safety switch LED at 10Hz */
+		safety_check_button();
+
+		/* Make the safety button flash anyway, no matter if it's used or not. */
+		flash_safety_button();
+
+		safety.timestamp = hrt_absolute_time();
+		safety.safety_switch_available = true;
+		safety.safety_off = _safety_btn_off;
+
+		/* lazily publish the safety status */
+		if (_to_safety != nullptr) {
+			orb_publish(ORB_ID(safety), _to_safety, &safety);
+
+		} else {
+			int instance;
+			_to_safety = orb_advertise_multi(ORB_ID(safety), &safety, &instance, ORB_PRIO_DEFAULT);
+		}
+	}
+
+	if (should_exit()) {
+		exit_and_cleanup();
+
+	} else {
+		/* schedule next cycle */
+		work_queue(HPWORK, &_work, (worker_t)&SafetyButton::cycle_trampoline, this, USEC2TICK(SCHEDULE_INTERVAL));
+	}
+}
+
+int SafetyButton::custom_command(int argc, char *argv[])
+{
+	return print_usage("unknown command");
+}
+
+int SafetyButton::print_usage(const char *reason)
+{
+	if (reason) {
+		PX4_WARN("%s\n", reason);
+	}
+
+	PRINT_MODULE_DESCRIPTION(
+		R"DESCR_STR(
+### Description
+This module is responsible for the safety button.
+
+)DESCR_STR");
+
+	PRINT_MODULE_USAGE_NAME("safety_button", "driver");
+	PRINT_MODULE_USAGE_COMMAND_DESCR("start", "Start the safety button driver");
+
+	return 0;
+}
+
+int SafetyButton::print_status()
+{
+#ifdef GPIO_BTN_SAFETY
+	if (!PX4_MFT_HW_SUPPORTED(PX4_MFT_PX4IO)) {
```

Comment:
This check can be removed now.

Reply:
@bkueng - If this is used on V5 the same binary is used on the the mini and it is needed.

Reply:
No, the module won't be running in that case, because of https://github.com/PX4/Firmware/pull/10119/files#diff-a461bff617e42003304db8e6ead04059R204.
`SafetyButton::print_status()` is only called if the module is running.

Reply:
ahh good point - I should have looked at the context.